go/vt/vtgate/executor.go

```diff
@@ -1494,8 +1493,9 @@ func (e *Executor) getPlan(vcursor *vcursorImpl, sql string, comments sqlparser.
 	return e.checkThatPlanIsValid(stmt, plan)
 }
 
-func (e *Executor) canNormalizeStatement(stmt sqlparser.Statement, qo iQueryOption, setVarComment string) bool {
-	return (e.normalize && sqlparser.CanNormalize(stmt)) ||
+func (e *Executor) canNormalizeStatement(stmt sqlparser.Statement, bindVars map[string]*querypb.BindVariable, qo iQueryOption, setVarComment string) bool {
+	// We do not normalize already prepared statements, by default
+	return (e.normalize && (*normalizePrepStmts || len(bindVars) == 0) && sqlparser.CanNormalize(stmt)) ||
```

Comment:
The point of extracting this into a function was breaking this logic into several `if`s instead of a single logical expression. I think that would be a worthy refactoring that would make this easier to follow.

Reply:
wildo

Reply:
Hey @vmg hopefully I understood your intention -- I broke up the components of this `if` into multiple `if` statements. Can break the first one down further if you like LMK if you had something else in mind here.